force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,115 @@
+public with sharing class OpportunityTriggerHandler {
+    public static void validateAmount(List<Opportunity> opps) {
+        for (Opportunity opp : opps) {
+            if (opp.Amount < 5000) {
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+    }
+
+    public static void setPrimaryContact(List<Opportunity> newOpps) {
+        Set<Id> accountIds = new Set<Id>();
+        for (Opportunity opp : newOpps) {
+            accountIds.add(opp.AccountId);
+        }
+        
+        Map<Id, Contact> contacts = new Map<Id, Contact>([SELECT Id, FirstName, AccountId FROM Contact WHERE AccountId IN :accountIds AND Title = 'CEO' ORDER BY FirstName ASC]);
+        Map<Id, Contact> accountIdToContact = new Map<Id, Contact>();
+
+        for (Contact cont : contacts.values()) {
+            if (!accountIdToContact.containsKey(cont.AccountId)) {
+                accountIdToContact.put(cont.AccountId, cont);
+            }
+        }
+
+        for(Opportunity opp : newOpps){
+            if(opp.Primary_Contact__c == null){
+                if (accountIdToContact.containsKey(opp.AccountId)){
+                    opp.Primary_Contact__c = accountIdToContact.get(opp.AccountId).Id;
+                }
+            }
+        }
+    }
+
+    public static void setType(List<Opportunity> newOpps){
+        for(Opportunity opp : newOpps){
+            if (opp.Type == null){
+                opp.Type = 'New Customer';
+            }   
+        }  
+    }
+       
+    public static void addStageChangeToDescription(List<Opportunity> opps, Map<Id, Opportunity> oldOppMap) {
+        for (Opportunity opp : opps){
+            Opportunity oldOpp = oldOppMap.get(opp.Id);
+            if (opp.StageName != oldOpp.StageName){
+                opp.Description += '\n Stage Change:' + opp.StageName + ':' + DateTime.now().format();
+            }                
+        }
+    }    
+     
+    public static void validateDelete(List<Opportunity> opps){
+        for (Opportunity oldOpp : opps){
+            if (oldOpp.IsClosed){
+                oldOpp.addError('Cannot delete closed opportunity');
+            }
+        }
+    }
+
+    public static void createTaskForPrimaryContact(List<Opportunity> opps){
+        List<Task> tasks = new List<Task>();
+            for (Opportunity opp : opps){
+                Task tsk = new Task();
+                tsk.Subject = 'Call Primary Contact';
+                tsk.WhatId = opp.Id;
+                tsk.WhoId = opp.Primary_Contact__c;
+                tsk.OwnerId = opp.OwnerId;
+                tsk.ActivityDate = Date.today().addDays(3);
```

Comment:
Try this with the date leteral for practice
https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select_dateformats.htm